components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -0,0 +1,94 @@
+import CheckBox from "../components/CheckBox";
+import { getGitpodService } from "../service/service";
+import { getCurrentTeam, TeamsContext } from "../teams/teams-context";
+import { PageWithSubMenu } from "../components/PageWithSubMenu";
+import PillLabel from "../components/PillLabel";
+
+export function getProjectSettingsMenu(project?: Project, team?: Team) {
+    const teamOrUserSlug = !!team ? 't/' + team.slug : 'projects';
+    return [
+        {
+            title: 'Configuration',
+            link: [`/${teamOrUserSlug}/${project?.slug || project?.name}/configure`],
+        },
+        {
+            title: 'Project',
+            link: [`/${teamOrUserSlug}/${project?.slug || project?.name}/settings`],
+        },
+    ];
+}
+
+export default function () {
+    const location = useLocation();
+    const { teams } = useContext(TeamsContext);
+    const team = getCurrentTeam(location, teams);
+    const match = useRouteMatch<{ team: string, resource: string }>("/(t/)?:team/:resource");
+    const projectSlug = match?.params?.resource;
+    const [ project, setProject ] = useState<Project | undefined>();
+
+    const [ isLoading, setIsLoading ] = useState<boolean>(true);
+    const [ isIncrementalPrebuildsEnabled, setIsIncrementalPrebuildsEnabled ] = useState<boolean>(false);
+
+    useEffect(() => {
+        if (!teams || !projectSlug) {
+            return;
+        }
+        (async () => {
+            const projects = (!!team
+                ? await getGitpodService().server.getTeamProjects(team.id)
+                : await getGitpodService().server.getUserProjects());
+
+            // Find project matching with slug, otherwise with name
+            const project = projectSlug && projects.find(p => p.slug ? p.slug === projectSlug : p.name === projectSlug);
+            if (!project) {
+                return;
+            }
+            setProject(project);
+        })();
+    }, [ projectSlug, team, teams ]);
+
+    useEffect(() => {
+        if (!project) {
+            return;
+        }
+        setIsLoading(false);
+        setIsIncrementalPrebuildsEnabled(!!project.settings?.useIncrementalPrebuilds);
+    }, [ project ]);
+
+    const toggleIncrementalPrebuilds = async () => {
+        if (!project) {
+            return;
+        }
+        setIsLoading(true);
+        try {
+            await getGitpodService().server.updateProjectPartial({
+                id: project.id,
+                settings: {
+                    useIncrementalPrebuilds: !isIncrementalPrebuildsEnabled,
+                }
+            });
+            setIsIncrementalPrebuildsEnabled(!isIncrementalPrebuildsEnabled);
+        } finally {
+            setIsLoading(false);
+        }
+    }
+
+    return <PageWithSubMenu subMenu={getProjectSettingsMenu(project, team)} title="Settings" subtitle={`Manage settings for your project ${project?.name}`}>
+        <h3>Incremental Prebuilds</h3>
+        <CheckBox
```

Comment:
**issue**: Could we use `max-w-2xl` for the checbox text here to limit the width of this element as we do with similar sections like Account settings?

**fyi**: The grid layout from earlier mockups is not accurately translated in the dashboard yet but will become more relevant once we pick up issues related to responsive design of the dashboard. Posting below an early draft of the responsive design of the grid. Using `max-w-2xl` here is a workaround to approach as much as possible the designs we've been using so far to create visual balance when needed. Feedback is welcome!

| Grid layout in large, medium, small and extra small viewports |
|-|
| <img width="1705" alt="Screenshot 2021-12-09 at 10 27 52 PM" src="https://user-images.githubusercontent.com/120486/145470842-f756fb77-a684-4a61-bf20-1fe06b1a4f49.png"> |

This is also an issue with other parts of the product, see [relevant discussion](https://gitpod.slack.com/archives/C02CBKZEVS8/p1638795733242600?thread_ts=1638366112.229500&cid=C02CBKZEVS8) (internal). Cc @gitpod-io/engineering-ide